### PR TITLE
[UXE-5585] feat: migrate domains to v4

### PR DESCRIPTION
**Pull request by pauloSF0**
### What is the new behavior introduced by this PR?
[UXE-5585] feat: migrate domains to v4
### Does this PR introduce breaking changes?
- [x] No
- [ ] Yes 

### Does this PR introduce UI changes? Add a video or screenshots here.

https://github.com/user-attachments/assets/f97c8fd0-a96b-4d7b-bd63-615080cff4fa

### Does it have a link on Figma?

<hr />

#### Make sure your pull request fits the checklist below (when applicable):

- [x] The issue title follows the format: [ISSUE_CODE] TYPE: TITLE
- [x] Commits are tagged with the right word (feat, test, refactor, etc)
- [ ] Application responsiveness was tested to different screen sizes
- [x] Code is formatted and linted
- [x] Tags are added to the PR

#### These changes were tested on the following browsers:
- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Safari

[UXE-5585]: https://aziontech.atlassian.net/browse/UXE-5585?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by peterpaulo-azion — APPROVED** [review]

File: src/services/domains-services/v4/list-domain-service.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+const listEdgeApplications = async () => {
+  const edgeApplications = await listEdgeApplicationsService({})
+
+  return edgeApplications.map((edgeApp) => ({ name: edgeApp.name, id: edgeApp.id }))
+}
+
+const getEdgeApplication = (edgeApplications, edge_application_id) => {
+  const edgeApplication = edgeApplications.filter((edgeapp) => edgeapp.id === edge_application_id)
+
+  if (edgeApplication.length > 0) {
```
Comment (lucasmendes21) [inline src/services/domains-services/v4/list-domain-service.js:66]: pode usar apenas o if (edgeApplication.length)
Como o zero é negativo ele já vai verificar.

File: src/services/domains-services/v4/list-domain-service.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    return edgeApplication[0].name
```
Comment (lucasmendes21) [inline src/services/domains-services/v4/list-domain-service.js:67]: edgeApplication[0]?.name
Verificar para evita erros de execução

File: src/services/domains-services/v4/list-domain-service.js (hunk 1 above)
Comment (pauloSF0) [reply src/services/domains-services/v4/list-domain-service.js:66]: okay

File: src/services/domains-services/v4/list-domain-service.js (hunk 2 above)
Comment (pauloSF0) [reply src/services/domains-services/v4/list-domain-service.js:67]: okay

**Review by lucasmendes21 — APPROVED** [review]

**Pull request merged**